sycl/source/detail/scheduler/commands.cpp

```diff
@@ -1956,8 +1956,13 @@ ExecCGCommand::ExecCGCommand(
               Dependencies),
       MEventNeeded(EventNeeded), MCommandGroup(std::move(CommandGroup)) {
   if (MCommandGroup->getType() == detail::CGType::CodeplayHostTask) {
-    MEvent->setSubmittedQueue(
-        static_cast<detail::CGHostTask *>(MCommandGroup.get())->MQueue);
+    const auto &SubmitQueue =
+        static_cast<detail::CGHostTask *>(MCommandGroup.get())->MQueue;
+    MEvent->setSubmittedQueue(SubmitQueue);
+    // Initialize host profiling info if the queue has profiling enabled.
+    if (SubmitQueue && SubmitQueue->MIsProfilingEnabled) {
+      MEvent->initHostProfilingInfo();
+    }
```

Comment:
```suggestion
    queue_impl *SubmitQueue = getQueue();
    if (!SubmitQueue)
      return;
      
    MEvent->setSubmittedQueue(SubmitQueue->weak_from_this());
    if (SubmitQueue->MIsProfilingEnabled)
      MEvent->initHostProfilingInfo();
```

to align with ongoing smart ptr refactorings. Would this work?